rust-runtime/aws-smithy-http-server-python/src/types.rs

```diff
@@ -288,22 +289,20 @@ impl futures::stream::Stream for ByteStream {
     type Item = Result<Bytes, aws_smithy_http::byte_stream::Error>;
 
     fn poll_next(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
-        let mut stream = self.0.lock();
-        Pin::new(&mut *stream).poll_next(cx)
-    }
-
-    fn size_hint(&self) -> (usize, Option<usize>) {
-        self.0.lock().size_hint()
+        let stream = self.0.lock();
+        tokio::pin!(stream);
```

Comment:
I need to really dig into tokio. I did not know about `tokio::pin!`. Very nice @weihanglo !!

Reply:
I myself need to really dig into `std::pin::Pin` as well 😆